src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,53 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+    private static final int NOT_FOUND_INDEX = -1;
+    private final K[] keyArr;
+    private final V[] valueArr;
+    private int size;
+
+    public StorageImpl() {
+        keyArr = (K[]) new Object[MAX_SIZE];
+        valueArr = (V[]) new Object[MAX_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = indexOfKey(key);
+        if (index != NOT_FOUND_INDEX) {
+            valueArr[index] = value;
+            size--;
```

Comment:
why do you decrement size?

UPD: I see now, what Andrii meant is using the `return`s instead of complications in the `if` `else`s
```suggestion
            return;
```
next time feel free to ask questions if you think that some decision or comment is strange or if you think you do not completely understand it